### PR TITLE
Use built-in check-migrations command for Airflow>=2 in helm chart

### PR DESCRIPTION
**Pull request by ashb**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: chart/templates/_helpers.yaml [hunk 1]
```diff
@@ -542,7 +542,11 @@ Create the name of the cleanup service account to use
 {{- end -}}
 
 {{ define "wait-for-migrations-command" }}
-  {{/* From Airflow 2.0.0 this can become [airflow, db, check-migrations] */}}
+  {{- if semverCompare ">=2.0.0" .Values.airflowVersion }}
```
Comment (kaxil) [inline chart/templates/_helpers.yaml:545]: Should we add a simple test like the following:

```diff
diff --git a/chart/tests/test_webserver.py b/chart/tests/test_webserver.py
index 329e6117d..6f92af2d6 100644
--- a/chart/tests/test_webserver.py
+++ b/chart/tests/test_webserver.py
@@ -137,6 +137,21 @@ class WebserverDeploymentTest(unittest.TestCase):
             "image": "test-registry/test-repo:test-tag",
         } == jmespath.search("spec.template.spec.containers[-1]", docs[0])

+    @parameterized.expand(
+        [
+            ("2.0.0", ["airflow", "db", "check-migrations"]),
+            ("2.1.0", ["airflow", "db", "check-migrations"]),
+        ],
+    )
+    def test_wait_for_migration_airflow_version(self, airflow_version, expected_arg):
+        docs = render_chart(
+            values={
+                "airflowVersion": airflow_version,
+            },
+            show_only=["templates/webserver/webserver-deployment.yaml"],
+        )
+        assert expected_arg == jmespath.search("spec.template.spec.initContainers[0].args", docs[0])
+
     def test_should_add_extra_init_containers(self):
         docs = render_chart(
             values={
```

**Comment by ashb** [comment]
Well _some_ of the Helm chart tests passed, that's promising.

**Comment by ashb** [comment]
```  E       AssertionError: Could not enable DAG: {'detail': None, 'status': 404, 'title': "Dag with id: 'example_kubernetes_executor_config' not found", 'type': 'http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/stable-rest-api-ref.html#section/Errors/NotFound'}```

Curious. I wonder why only some of the helm chart matrix failed with that.

**Comment by kaxil** [comment]
> ` E AssertionError: Could not enable DAG: {'detail': None, 'status': 404, 'title': "Dag with id: 'example_kubernetes_executor_config' not found", 'type': 'http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/stable-rest-api-ref.html#section/Errors/NotFound'}`
> 
> Curious. I wonder why only some of the helm chart matrix failed with that.

Oh !!! 🤦  We deleted that DAG and added everything `example_kubernetes_executor`

**Comment by kaxil** [comment]
Need:

```
diff --git a/kubernetes_tests/test_kubernetes_executor.py b/kubernetes_tests/test_kubernetes_executor.py
index 1ca3848fe..61030d1b9 100644
--- a/kubernetes_tests/test_kubernetes_executor.py
+++ b/kubernetes_tests/test_kubernetes_executor.py
@@ -25,7 +25,7 @@ from kubernetes_tests.test_base import EXECUTOR, TestBase
 @pytest.mark.skipif(EXECUTOR != 'KubernetesExecutor', reason="Only runs on KubernetesExecutor")
 class TestKubernetesExecutor(TestBase):
     def test_integration_run_dag(self):
-        dag_id = 'example_kubernetes_executor_config'
+        dag_id = 'example_kubernetes_executor'
         dag_run_id, execution_date = self.start_job_in_kubernetes(dag_id, self.host)
         print(f"Found the job with execution_date {execution_date}")

@@ -48,7 +48,7 @@ class TestKubernetesExecutor(TestBase):
         )

     def test_integration_run_dag_with_scheduler_failure(self):
-        dag_id = 'example_kubernetes_executor_config'
+        dag_id = 'example_kubernetes_executor'

         dag_run_id, execution_date = self.start_job_in_kubernetes(dag_id, self.host)
```

But we can create a separate PR for that

**Comment by ashb** [comment]
Turns out living with broken tests is bad -- who'da thunk it!?

**Comment by ashb** [comment]
https://github.com/apache/airflow/pull/19680

File: chart/templates/_helpers.yaml (hunk 1 above)
Comment (ashb) [reply chart/templates/_helpers.yaml:545]: Test added (and one for < 2.0 condition too.)

**Review by kaxil — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
Nice!. Just got back to my PC :)

**Pull request merged**